libs/community/langchain_community/chat_models/githubllm.py

```diff
@@ -0,0 +1,314 @@
+import os
+import logging
+import time
+from typing import Any, Dict, List, Mapping, Optional, Iterator, Union, AsyncIterator
+import asyncio
+from concurrent.futures import ThreadPoolExecutor
+
+import requests
+from langchain_core.callbacks import CallbackManagerForLLMRun, AsyncCallbackManagerForLLMRun
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.messages import BaseMessage, AIMessage, HumanMessage, SystemMessage
+from langchain_core.outputs import ChatResult, ChatGeneration, ChatGenerationChunk
+from langchain_core.pydantic_v1 import Extra, Field, root_validator, SecretStr
+
+logger = logging.getLogger(__name__)
+
+def run_in_executor(executor, func, *args, **kwargs):
+    loop = asyncio.get_event_loop()
+    return loop.run_in_executor(executor, func, *args, **kwargs)
+
+def _convert_message_to_dict(message: BaseMessage) -> dict:
+    if isinstance(message, SystemMessage):
+        return {"role": "system", "content": message.content}
+    elif isinstance(message, HumanMessage):
+        return {"role": "user", "content": message.content}
+    elif isinstance(message, AIMessage):
+        return {"role": "assistant", "content": message.content}
+    else:
+        raise ValueError(f"Got unknown type {message}")
+
+def _convert_dict_to_message(message_dict: Mapping[str, Any]) -> BaseMessage:
+    role = message_dict["role"]
+    content = message_dict.get("content", "")
+    if role == "user":
+        return HumanMessage(content=content)
+    elif role == "assistant":
+        return AIMessage(content=content)
+    elif role == "system":
+        return SystemMessage(content=content)
+    else:
+        raise ValueError(f"Got unknown role {role}")
+
+class GithubLLM(BaseChatModel):
```

Comment:
We should rename this to `Chat____` to match convention for other models (e.g., `ChatGithub`).

I'm wondering if `Github` is the right name for this integration, though, given that the endpoint is https://models.inference.ai.azure.com/. My first thought would be `azure_ai_inference` / `ChatAzureAIInference` or similar, given the naming in the docs [here](https://learn.microsoft.com/en-us/azure/ai-studio/reference/reference-model-inference-api). Let me know if I'm looking at the wrong product.

Reply:
so i'm calling it Github because the product page is here : https://github.com/marketplace/models and this implementation looks for a github PAT first then falls back on azure ...

that said, i'm totally agnostic about this , we could even entirely drop github PAT support to favor azure , although using github PAT makes using the object completely free , which is quite nice :-)